presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/EncryptionHelper.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import org.apache.commons.codec.digest.Crypt;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.codec.digest.Md5Crypt;
+import org.mindrot.jbcrypt.BCrypt;
+
+import static java.util.Base64.getEncoder;
+
+public class EncryptionHelper
+{
+    private EncryptionHelper()
+    {
+    }
+
+    public static boolean checkMD5Password(String inputPassword, String hashedPassword)
+    {
+        try {
+            return hashedPassword.equals(Md5Crypt.apr1Crypt(inputPassword, hashedPassword));
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
Why do you catch these? I think you should not do that, it may hide some bugs in the code.